lib/trackler/implementation.rb

```diff
@@ -13,11 +13,11 @@ class Implementation
 
     attr_reader :track_id, :repo, :problem, :root, :file_bundle
     attr_writer :files
-    def initialize(track_id, repo, problem, root)
-      @track_id = track_id
-      @repo = repo
+    def initialize(track, problem)
```

Comment:
Glad you like it. At first I was feeling gross about adding yet more data for the implementation to rely on, but once I realized that every single piece of data was actually a part of track, it suddenly felt much cleaner.